docs/geode-native-docs-cpp/serialization/cpp-serialization/pdxserializable-interface.html.md.erb

```diff
@@ -27,88 +27,60 @@ A domain class should serialize and de-serialize all its member fields in the sa
+    }
+    ```
 
-private:
-    uint32_t m_id;
-    char* m_str;
+    In your `fromData` implementation, use the same name as you did in `toData` and call the read operations in the same order as you called the write operations in your `toData` implementation.
 
-public:
-    PdxObject(){};
-    PdxObject(uint32_t id, char* str);
-    virtual ~PdxObject();
+4.  Optionally, program your domain object's `hashCode` and equality methods. When you do so, you can optimize those methods by specifying the _identity fields_
+    to be used in comparisons. 
+    <br /><br />
+    If no fields are set as identity fields, then all fields will be used in `hashCode` and equality checks, so marking identity fields improves the efficiency
+    of hashing and equality operations. 
+    <br /><br />
+    Use the `markIdentityField` method to indicate that the given field name should be included in `hashCode` and equality checks of this object on a server.
+    It is important that the fields used by your equality method and `hashCode` implementations are the same fields that you mark as identity fields.
+    <br /><br />
+    Expanding the code sample from the description of the `toData` method, above:
 
-    uint32_t getID() {
-        return m_id;
-    }
+    ``` cpp
+    void Order::toData(PdxWriter& pdxWriter) const {
+      pdxWriter.writeInt(ORDER_ID_KEY_, order_id_);
+      pdxWriter.markIdentityField(ORDER_ID_KEY_);
 
-    char* getStr(){
-        return m_str;
-    }
+      pdxWriter.writeString(NAME_KEY_, name_);
+      pdxWriter.markIdentityField(NAME_KEY_);
 
-    virtual void toData(PdxWriterPtr pw) const;
-    virtual void fromData(PdxReaderPtr pr);
-    CacheableStringPtr toString() const;
-    virtual char* getClassName() const;
-    static Cacheable* createDeserializable() {
-        return new PdxObject();
+      pdxWriter.writeShort(QUANTITY_KEY_, quantity_);
+      pdxWriter.markIdentityField(QUANTITY_KEY_);
     }
-};
-
-PdxObject::PdxObject(uint32_t i, char* str) {
-    m_id = i;
-    m_str = str;
-}
-
-PdxObject::~PdxObject() {
-}
-
-void PdxObject::toData( PdxWriterPtr pw ) const {
-    pw->writeInt("id", m_id);
-       pw->markIdentityField("id");
-    pw->writeString("str", m_str);
-}
-
-void  PdxObject::fromData( PdxReaderPtr pr )
-{
-    m_id = pr->readInt("id");
-    m_str = pr->readString("str");
-}
-
-char* getClassName() const{
-{
-    return "com.example.PdxType";
-}
-
-CacheableStringPtr PdxObject::toString() const {
-       char idbuf[1024];
-       sprintf(idbuf,"PdxObject: [ ID=%d ]",m_id);
-       return CacheableString::create( idbuf );
-}
-```
-
+    ```
+    The fields that are marked as identity fields are used to generate the `hashCode` and equality
+    methods of PdxInstance. Because of this, the identity fields should themselves either be primitives,
+    or implement `hashCode` and equals.  The identity fields should be marked after they are written
+    using a `write*` method.
```

Comment:
Line 85-86, I'd prefer to see a different wording that directly states that we are ordering the function calls. Perhaps something like:
"Invoke the `markIdentityField` function directly following its `write*` function.

I would do the same as surrounding prose, but I think of C++ as having functions, while Java has methods. Probably no one cares which term you use.

Reply:
Good suggestions, @karensmolermiller. I checked in a revision - please give it a second look.